### ISSUE TITLE
Propper support to NavController

### ISSUE THREAD

**Issue by ismaeldivita** [ismaeldivita/chip-navigation-bar#20]

**Comment by NIkhilbadveli** [comment]
I pulled off a small trick to do this. You need to do it this way:

1.  Create a bottomnavigation view in xml file like this and set its visibility to "Gone".
` <com.google.android.material.bottomnavigation.BottomNavigationView
        android:id="@+id/bottomNav"
        android:layout_width="match_parent"
        android:layout_height="wrap_content"
        android:visibility="gone"
        app:menu="@menu/bottom_menu" />`
2. And then do it like this in your activity file.
`val navController: NavController = Navigation.findNavController(this, R.id.fragment)
        val bottomNavigationView: BottomNavigationView = findViewById(R.id.bottomNav)
        val chipNavigationBar: ChipNavigationBar = findViewById(R.id.chip_nav)

        chipNavigationBar.setOnItemSelectedListener { itemId ->
            bottomNavigationView.selectedItemId = itemId
        }

        NavigationUI.setupWithNavController(bottomNavigationView, navController)`


What I'm doing is making android think that there's a bottomnavigation that's changing it's state whenever we click on our chip navigator here.

Try this and let me know if it solves your problem.

**Comment by jafar260698** [comment]
it is not working for me

**Comment by shivjyotigarai** [comment]
> I pulled off a small trick to do this. You need to do it this way:
> 
> 1. Create a bottomnavigation view in xml file like this and set its visibility to "Gone".
>    ` <com.google.android.material.bottomnavigation.BottomNavigationView android:id="@+id/bottomNav" android:layout_width="match_parent" android:layout_height="wrap_content" android:visibility="gone" app:menu="@menu/bottom_menu" />`
> 2. And then do it like this in your activity file.
>    `val navController: NavController = Navigation.findNavController(this, R.id.fragment) val bottomNavigationView: BottomNavigationView = findViewById(R.id.bottomNav)    val chipNavigationBar: ChipNavigationBar = findViewById(R.id.chip_nav)     chipNavigationBar.setOnItemSelectedListener { itemId -> bottomNavigationView.selectedItemId = itemId }     NavigationUI.setupWithNavController(bottomNavigationView, navController)`
> 
> What I'm doing is making android think that there's a bottomnavigation that's changing it's state whenever we click on our chip navigator here.
> 
> Try this and let me know if it solves your problem.

@NIkhilbadveli can u plz write the code in activity file (2) in java plz

**Comment by NIkhilbadveli** [comment]
It's pretty straightforward... maybe try online for converting kotlin code to java

**Comment by shivjyotigarai** [comment]
Just facing issue with this line...`chipNavigationBar.setOnItemSelectedListener { itemId -> bottomNavigationView.selectedItemId = itemId }`.   @NIkhilbadveli

**Comment by shivjyotigarai** [comment]
@NIkhilbadveli which itemId is referred above

**Comment by NIkhilbadveli** [comment]
.setOnItemSelectedListener(new ChipNavigationBar.OnItemSelectedListener() {
    @Override
    public void onItemSelected(int id) {
        // TODO
    }
});

Try it this way.

**Comment by shivjyotigarai** [comment]
> .setOnItemSelectedListener(new ChipNavigationBar.OnItemSelectedListener() {
> @override
> public void onItemSelected(int id) {
> // TODO
> }
> });
> 
> Try it this way.

thx

**Comment by NIkhilbadveli** [comment]
itemId is the id of the selection in chip navigation bar. I'm taking that and passing it to bottomnavigationview, which actually does the navigation.

**Comment by Suret1** [comment]
> I pulled off a small trick to do this. You need to do it this way:
> 
> 1. Create a bottomnavigation view in xml file like this and set its visibility to "Gone".
>    ` <com.google.android.material.bottomnavigation.BottomNavigationView android:id="@+id/bottomNav" android:layout_width="match_parent" android:layout_height="wrap_content" android:visibility="gone" app:menu="@menu/bottom_menu" />`
> 2. And then do it like this in your activity file.
>    `val navController: NavController = Navigation.findNavController(this, R.id.fragment) val bottomNavigationView: BottomNavigationView = findViewById(R.id.bottomNav)    val chipNavigationBar: ChipNavigationBar = findViewById(R.id.chip_nav)     chipNavigationBar.setOnItemSelectedListener { itemId -> bottomNavigationView.selectedItemId = itemId }     NavigationUI.setupWithNavController(bottomNavigationView, navController)`
> 
> What I'm doing is making android think that there's a bottomnavigation that's changing it's state whenever we click on our chip navigator here.
> 
> Try this and let me know if it solves your problem.

hi. should i do this?
it didn't work for me too. I will be glad if you help me

![Screenshot 2021-01-22 185505](https://user-images.githubusercontent.com/19667419/105506461-82fecb80-5ce3-11eb-810c-ea3d2aee4c7f.jpg)

![2](https://user-images.githubusercontent.com/19667419/105506651-b3df0080-5ce3-11eb-96bf-47497076d814.jpg)

**Comment by DevLuke67** [comment]
> Just facing issue with this line...`chipNavigationBar.setOnItemSelectedListener { itemId -> bottomNavigationView.selectedItemId = itemId }`. @NIkhilbadveli

Thanks!

**Comment by OmarELRayes** [comment]
but how do you handle "back" with it , chip navigation view doesn't update its items

**Comment by NIkhilbadveli** [comment]
If you set the chip navigation item whenever there's a navigation change, then it should work. `//Changing chip navigation selection on change
        navController.addOnDestinationChangedListener { _, destination, _ ->
            chipNavigationBar.setItemSelected(destination.id)
        }`

**Comment by tolgaerbas1** [comment]
Is there any update on Navigation support?

**Comment by saurabhthesuperhero** [comment]
So 

> If you set the chip navigation item whenever there's a navigation change, then it should work. `//Changing chip navigation selection on change navController.addOnDestinationChangedListener { _, destination, _ -> chipNavigationBar.setItemSelected(destination.id) }`

OMG thanks so  all the above worked properly as you mentioned.

**Comment by SB2318** [comment]
Yes, It is true that we can't set this bottom nav bar with NavController.
But we can use it with nav controller, and It is properly working for me.

I am sharing here the code snippet.

```
 binding.bottomNavBar.setOnItemSelectedListener {

            when(it){

                R.id.taskFragment->{
                    navController.navigate(R.id.taskFragment)
                }
                R.id.projectFragment->{

                    navController.navigate(R.id.projectFragment)
                }

                R.id.timeSheetFragment->{

                    navController.navigate(R.id.timeSheetFragment)
                }

                R.id.moreFragment->{

                    navController.navigate(R.id.moreFragment)
                }
            }
        }
```

This is a trick that's why I keep the menu_itemId and destinationFragmentId's are same.

**Comment by alpeshSolanki7271** [comment]
when I back press at last position ..fragment changed but icon not change..
